In Gentoo, we don't yet usually have HTTP3 for curl either.

https://src.fedoraproject.org/rpms/python-pycurl/raw/rawhide/f/0002-python-pycurl-7.45.2-disable-test_http_version_3.patch

From 6c55e6822a064edb8dd6dfe1ee21080f4f30b5cf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Luk=C3=A1=C5=A1=20Zaoral?= <lzaoral@redhat.com>
Date: Tue, 7 Mar 2023 16:32:19 +0100
Subject: [PATCH] tests: temporarily disable test_http_version_3

HTTP3 support in curl is only experimental at the moment and is disabled
in the corresponding Fedora package.
--- a/tests/option_constants_test.py
+++ b/tests/option_constants_test.py
@@ -508,10 +508,6 @@ class OptionConstantsSettingTest(unittest.TestCase):
     def test_http_version_2prior_knowledge(self):
         self.curl.setopt(self.curl.HTTP_VERSION, self.curl.CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE)
 
-    @util.min_libcurl(7, 66, 0)
-    def test_http_version_3(self):
-        self.curl.setopt(self.curl.HTTP_VERSION, self.curl.CURL_HTTP_VERSION_3)
-
     @util.min_libcurl(7, 21, 5)
     def test_sockopt_constants(self):
         assert self.curl.SOCKOPT_OK is not None
-- 
2.39.2


